sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/EmptyLineRegexCheck.kt

```diff
@@ -0,0 +1,258 @@
+        return (when (val parent = callExpression.parent?.parent) {
+            is KtProperty -> {
+                parent.usages()
+                    .mapNotNull { getStringInRegexFind(it, bindingContext) }
+                    .filter { it.canBeEmpty(bindingContext) }
+            }
+            is KtDotQualifiedExpression -> {
+                getStringInRegexFind(parent.selectorExpression!!, bindingContext)
+                    ?.let { if (it.canBeEmpty(bindingContext)) listOf(it) else emptyList()} ?: emptyList()
+            }
+            else -> {
+                emptyList()
+            }
+        }) to (callExpression.parent as? KtExpression)?.getResolvedCall(bindingContext)?.getReceiverExpression()
+    }
+
+    private fun getStringInMatcherFind(ref: KtElement, bindingContext: BindingContext): KtExpression? {
+        val resolvedCall = (ref.parent as? KtExpression)?.getResolvedCall(bindingContext) ?: return null
+        if (!(resolvedCall matches PATTERN_MATCHER)) return null
+        val preParent = ref.parent.parent ?: return null
+
+        return when (preParent) {
+            is KtExpression -> if (preParent.getResolvedCall(bindingContext) matches PATTERN_FIND) extractArgument(resolvedCall) else null
+            is KtProperty -> if (preParent.usages().any { it.getParentResolvedCall(bindingContext) matches PATTERN_FIND }) {
+                extractArgument(resolvedCall)
+            } else null
+            else -> null
+        }
+    }
+
+    private fun getStringInRegexFind(ref: KtExpression, bindingContext: BindingContext): KtExpression? {
+        val resolvedCall = (ref.parent as? KtExpression)?.getResolvedCall(bindingContext) ?: return null
+        return if (resolvedCall matches REGEX_FIND) extractArgument(resolvedCall) else null
+    }
+    
+    private fun extractArgument(resolvedCall: ResolvedCall<out CallableDescriptor>): KtExpression? {
+        val arguments = resolvedCall.valueArgumentsByIndex
+        return if (arguments == null
+            || arguments.isEmpty()
+            || arguments[0] == null
+            || arguments[0] !is ExpressionValueArgument
+        ) null
+        else (arguments[0] as ExpressionValueArgument).valueArgument?.getArgumentExpression()
+    }
+
+    private fun KtExpression.canBeEmpty(bindingContext: BindingContext): Boolean =
+        when (val deparenthesized = this.deparenthesize()) {
+            is KtNameReferenceExpression -> {
+                val runtimeStringValue = deparenthesized.predictRuntimeStringValue(bindingContext)
+                runtimeStringValue?.isEmpty()
+                    ?: deparenthesized.findAllUsages {
+                        (it.parent as? KtExpression)?.getResolvedCall(bindingContext) matches STRING_IS_EMPTY
+                    }.isEmpty()
+            }
+            else ->
+                predictRuntimeStringValue(bindingContext)?.isEmpty() ?: false
+        }
```

Comment:
It took me a little to understand what this method does and what it looks for - WDYT of a (KDoc) comment to give a very brief explanation of what this method does (and potentially how)?